testsuite/MDAnalysisTests/core/test_atom.py

```diff
@@ -22,131 +22,125 @@
-        self.atom.mass = 13
-        assert_equal(self.atom.mass, 13)
-
-    def test_setting_attributes_charge(self):
-        self.atom.charge = 6
-        assert_equal(self.atom.charge, 6)
-
-    def test_attributes_positions(self):
-        a = self.atom
+    """Set up the standard AdK system in implicit solvent."""
+
+    @staticmethod
+    @pytest.fixture()
+    def universe():
+        return mda.Universe(PSF, DCD)
+
+    @staticmethod
+    @pytest.fixture()
+    def atom(universe):
+        # Leu67:CG
+        return universe.atoms[1000]
+
+    def test_attributes_names(self, atom):
+        a = atom
+        assert a.name == 'CG'
+        assert a.resname == 'LEU'
+
+    def test_setting_attribute_name(self, atom):
+        atom.name = 'AA'
+        assert atom.name == 'AA'
+
+    def test_setting_attribute_type(self, atom):
+        atom.type = 'Z'
+        assert atom.type == 'Z'
+
+    def test_setting_attribute_mass(self, atom):
+        atom.mass = 13
+        assert atom.mass == 13
+
+    def test_setting_attributes_charge(self, atom):
+        atom.charge = 6
+        assert atom.charge == 6
+
+    def test_attributes_positions(self, atom):
+        known_pos = np.array([3.94543672, -12.4060812, -7.26820087], dtype=np.float32)
+        a = atom
         # new position property (mutable)
-        assert_almost_equal(a.position, self.known_pos)
+        assert_almost_equal(a.position, known_pos)
         pos = a.position + 3.14
         a.position = pos
         assert_almost_equal(a.position, pos)
 
-    def test_atom_selection(self):
-        asel = self.universe.select_atoms('atom 4AKE 67 CG').atoms[0]
-        assert_equal(self.atom, asel)
+    def test_atom_selection(self, universe, atom):
+        asel = universe.select_atoms('atom 4AKE 67 CG').atoms[0]
+        assert atom == asel
 
-    def test_hierarchy(self):
-        u = self.universe
-        a = self.atom
-        assert_equal(a.segment, u.s4AKE)
-        assert_equal(a.residue, u.residues[66])
+    def test_hierarchy(self, universe, atom):
+        u = universe
+        a = atom
+        assert a.segment == u.s4AKE
+        assert a.residue == u.residues[66]
 
-    def test_bad_add(self):
+    def test_bad_add(self, atom):
         def bad_add():
-            return self.atom + 1
+            return atom + 1
 
-        assert_raises(TypeError, bad_add)
+        with pytest.raises(TypeError):
+            bad_add()
 
-    def test_add_AG(self):
-        ag = self.universe.atoms[:2]
+    def test_add_AG(self, universe, atom):
+        ag = universe.atoms[:2]
 
-        ag2 = self.atom + ag
+        ag2 = atom + ag
 
-        for at in [self.atom, ag[0], ag[1]]:
-            assert_equal(at in ag2, True)
+        for at in [atom, ag[0], ag[1]]:
+            assert at in ag2
 
-    def test_no_velo(self):
+    def test_no_velo(self, atom):
         def lookup_velo():
-            return self.atom.velocity
+            return atom.velocity
```

Comment:
Since you are using the context manager to test the exception, you do not need the nested function.